src/components/experiments/HeaderButtons/useOnForkButtonPressed.tsx

```diff
@@ -0,0 +1,64 @@
+import { useCallback, useEffect } from "react";
+import { api } from "~/utils/api";
+import { useExperiment, useHandledAsyncCallback } from "~/utils/hooks";
+import { signIn, useSession } from "next-auth/react";
+import { useRouter } from "next/router";
+import queryString from "query-string";
+
+export const useOnForkButtonPressed = () => {
+  const utils = api.useContext();
+  const router = useRouter();
+
+  const user = useSession().data;
+  const experiment = useExperiment();
+
+  const forkMutation = api.experiments.fork.useMutation();
+
+  const [onFork, isForking] = useHandledAsyncCallback(async () => {
+    if (!experiment.data?.id) return;
+    const forkedExperiment = await forkMutation.mutateAsync({ id: experiment.data.id });
+    await utils.experiments.list.invalidate();
```

Comment:
Do we need this? Adds latency and I actually think refreshing the experiment list right before going to the new one is more confusing than just going directly.

Reply:
No, I think you're right, we don't need it